pkg/sdk/helpers/resource_client.go

```diff
@@ -0,0 +1,279 @@
+}
+
+// ResourceEmptyError represents an error for a resource that was found, but is empty.
+type ResourceEmptyError ResourceError
+
+// Error returns a string representation of this error
+func (e *ResourceEmptyError) Error() string {
+	return fmt.Sprintf("found resource: '%s' %s, but it is empty", e.uri, getLocation(e.service, e.stage, e.project))
+}
+
+// ResourceUploadFailedError represents an error for a resource that could not be uploaded.
+type ResourceUploadFailedError struct {
+	ResourceError
+	message string
+}
+
+// Error returns a string representation of this error.
+func (e *ResourceUploadFailedError) Error() string {
+	return fmt.Sprintf("could not upload resource: '%s' %s: %s", e.uri, getLocation(e.service, e.stage, e.project), e.message)
+}
+
+// ResourceRetrievalFailedError represents an error for a resource that could not be retrieved because of an error.
+type ResourceRetrievalFailedError struct {
+	ResourceError
+	cause error
+}
+
+// Error returns a string representation of this error.
+func (e *ResourceRetrievalFailedError) Error() string {
+	return fmt.Sprintf("could not retrieve resource: '%s' %s: %v", e.uri, getLocation(e.service, e.stage, e.project), e.cause)
+}
+
+// Unwrap returns the cause of the ResourceRetrievalFailedError.
+func (e *ResourceRetrievalFailedError) Unwrap() error {
+	return e.cause
+}
+
+func getLocation(service string, stage string, project string) string {
+	var location string
+
+	if service != "" {
+		location += fmt.Sprintf(" for service '%s'", service)
+	}
+	if stage != "" {
+		location += fmt.Sprintf(" at stage '%s'", stage)
+	}
+	if project != "" {
+		location += fmt.Sprintf(" of project '%s'", project)
+	}
+
+	return strings.TrimLeft(location, " ")
+}
+
+// ResourceHelper is the default implementation for the ResourceClientInterface using a Keptn api.ResourcesV1Interface.
+type ResourceHelper struct {
+	client v2.ResourcesInterface
+}
+
+// NewResourceHelper creates a new ResourceHelper using a api.ResourcesV1Interface.
+func NewResourceHelper(client v2.ResourcesInterface) *ResourceHelper {
+	return &ResourceHelper{
+		client: client,
+	}
+}
+
+// GetResource tries to find the first instance of a given resource on service, stage or project level.
+func (rc *ResourceHelper) GetResource(ctx context.Context, project string, stage string, service string, resourceURI string) (string, error) {
+	var rnfErrorType *ResourceNotFoundError
+	if project != "" && stage != "" && service != "" {
+		keptnResourceContent, err := rc.GetServiceResource(ctx, project, stage, service, resourceURI)
+		if errors.As(err, &rnfErrorType) {
+			log.WithFields(
+				log.Fields{
+					projectFieldName:     project,
+					stageFieldName:       stage,
+					serviceFieldName:     service,
+					resourceURIFieldName: resourceURI,
+				}).Debug("Resource not available for service")
+		} else if err != nil {
+			return "", err
+		} else {
```

Comment:
that `else` can be removed

Reply:
@bacherfl it cannot be removed, as in this case we would return from the method execution and not check for stage or project resources.